src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,175 @@
+package duke;
+
+import duke.command.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents parser to parse user input
+ */
+public class Parser {
+
+    /** Regex of user input format */
+    public static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<command>\\S+)(\\s*)(?<args>.*)");
+
+    /**
+     * Returns command invoked by user with parsed arguments to execute.
+     * If command word given by user is invalid, HelpCommand is returned.
+     *
+     * @param userInput User input with command word and arguments to parse.
+     * @return Command to execute.
+     * @throws DukeException if user input does not match BASIC_COMMAND_FORMAT.
+     */
+    public static Command parse (String userInput) throws DukeException {
+        final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput);
+        if (!matcher.matches()) {
+            throw new DukeException("You better type sumthin or else..");
+        }
+
+        String command = matcher.group("command");
+        String args = matcher.group("args");
+
+        switch (command) {
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case DoneCommand.COMMAND_WORD:
+            return prepareDone(args);
+        case DeleteCommand.COMMAND_WORD:
+            return prepareDelete(args);
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(args);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(args);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(args);
+        case FindCommand.COMMAND_WORD:
+            return new FindCommand(args);
+        case HelpCommand.COMMAND_WORD: // Fallthrough
+        default:
+            return new HelpCommand();
+        }
+    }
+
+    /**
+     * Returns DoneCommand to execute.
+     *
+     * @param args User input argument to determine index of task to mark as done.
+     * @return DoneCommand to execute.
+     * @throws DukeException If there are no arguments given or argument is not an integer.
+     */
+    public static Command prepareDone(String args) throws DukeException {
```

Comment:
I think the naming of this method as well as the parameter could be made clearer, something like:
```suggestion
    public static Command returnDoneCommand(String indexOfTask) throws DukeException {
```